### DISCUSSION TITLE
Bug on json post réception

### DISCUSSION THREAD

**Discussion by fcabaud** [fastapi/fastapi#6688] [Questions] [answered]
### First check

* I added a very descriptive title to this issue.
* I used the GitHub search to find a similar issue and didn't find it.
* I searched the FastAPI documentation, with the integrated search.
* I already searched in Google "How to X in FastAPI" and didn't find any information.
* I already read and followed all the tutorial in the docs and didn't find an answer.
* I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
* I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
* I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).
* After submitting this, I commit to one of:
    * Read open issues with questions until I find 2 issues where I can help someone and add a comment to help there.
    * I already hit the "watch" button in this repository to receive notifications and I commit to help at least 2 people that ask questions in the future.
    * Implement a Pull Request for a confirmed bug.

### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import APIRouter, Request

from pydantic import BaseModel
fake_db = {}
class Item(BaseModel):
    action: str

app = APIRouter()

@app.post("/update_action/")
async def update_action(item: Item):
    print("coucou")
    item_dict = item.dict()
    if item.action:
        #price_with_tax = item.price + item.tax
        #item_dict.update({"price_with_tax": price_with_tax})
        print("found")
    return item_dict
```

The code for sending the Json data via a POST command is here.
It's done by clicking on sentence "Click for sending json data via fetch command"
```HTML
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01//EN">
<html>
<head>
  <title>page de debug</title>
  <!--<script src="fetch.js"></script>-->
</head>

<body>
    <p onclick="myFunction(this)">Click for sending json data via fetch command</p>

<script>
function myFunction(elmnt) {
  const data = { "action" : "myAction"};

console.log(JSON.stringify(data));

fetch("http://127.0.0.1:8000/update_action"  , {
    method: 'POST',
    mode: 'no-cors',
    headers: {
        'Content-Type' : 'application/json'
    },
    body: {
        body: JSON.stringify(data),
    }
})
    .then(function() {
        console.log("succesfully");
    }).catch(function() {
        console.log("error");
    });
}
</script>
</body>
</html>

```

### Description

* Open the browser and open the provided HTML code `/`.
* It display Click for sending json data via fetch command`.
* click on this sentence.
* Json is posted to python server.
* error is as follow:

`INFO:     127.0.0.1:43628 - "POST /update_action HTTP/1.1" 307 Temporary Redirect
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/.local/lib/python3.9/site-packages/fastapi/routing.py", line 184, in app
    body = await request.json()
  File "/home/.local/lib/python3.9/site-packages/starlette/requests.py", line 228, in json
    self._json = json.loads(body)
  File "/usr/lib/python3.9/json/__init__.py", line 346, in loads
    return _default_decoder.decode(s)
  File "/usr/lib/python3.9/json/decoder.py", line 337, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib/python3.9/json/decoder.py", line 355, in raw_decode
    raise JSONDecodeError("Expecting value", s, err.value) from None
json.decoder.JSONDecodeError: Expecting value: line 1 column 2 (char 1)`

### Environment

* OS: Ubuntu 21.04
* FastAPI Version: 0.65.2
* Python version: 3.9.5

**Comment by ArcLightSlavik** [comment, marked as answer]
Your endpoint is `@app.post("/update_action/")` but you are sending to `http://127.0.0.1:8000/update_action`

Change to `@app.post("/update_action")`.

I assume the 422 error after that is unrelated.

**Comment by me-v2** [comment]
this is not fastapi bugs, please close

**Comment by fcabaud** [comment]
In changed `@app.post("/update_action/")` into `@app.post("/update_action")` and the bug is still here
and "coucou" is still not displayed...

**Comment by fcabaud** [comment]
Well it seems, the javascript code is wrong.
